### PR TITLE
Full sync ask blocks only from other archival nodes

### LINKED ISSUE #929: Full sync should only ask blocks from other archival nodes
I believe it requires adding a feature bit for archival nodes, don't think we have that right now.

### PR DESCRIPTION
**Pull request by quentinlesceller**
Fix #929.

~~Just need to add the ```FULL_HIST``` internal check after discussion with @ignopeverell.~~ Done.

Not as elegant as I wanted it to be though.

Before starting the p2p server, in case of archive mode, we try to retrieve the full block 1. If it fails, we remove the ```FULL_HIST``` capability from the ```capabilities```.

### CODE REVIEW

**Comment by ignopeverell** [comment]
Looks like the `p2p` crate doesn't compile.

**Comment by quentinlesceller** [comment]
Weird, everything is working on my machine (and Travis?).

**Comment by ignopeverell** [comment]
Mmh not sure where I saw that, must have been confused sorry.

File: pool/src/pool.rs [hunk 1]
```diff
@@ -252,6 +252,11 @@ where
 		let stem_propagation = random <= self.config.dandelion_probability;
 		let mut will_stem = stem && stem_propagation;
 
+		// Track the case where a parent of a transaction is in stempool
```
Comment (ignopeverell) [inline pool/src/pool.rs]: Did you mean to commit that stuff too?

**Comment by ignopeverell** [comment]
Mmh not sure where I saw that, must have been confused sorry.

**Comment by quentinlesceller** [comment]
@ignopeverell Looks like I did a ```git checkout``` in the wrong branch :). Just merged branch from upstream.

**Pull request merged**